components/MainScreen.js

```diff
@@ -36,24 +36,36 @@ export default class MainScreen extends Component {
 		let driver = '';
 
 		// We should really just get rid of this whole query, and put a better version of this in MatchBanner.js
-		await store.collection('lots').onSnapshot( allLots => { // This query needs to be majorly changed..
-			allLots.docChanges().forEach(lot => {
-				driver = lot.doc.data().driverId;
-				//Not sure if needs another if statement but bid info should not changed unless its another bid
-				if (lot.doc.data().passengerId === auth.currentUser.email && lot.doc.data().driverId !== null) {
-					this.setState({showBid: true, offer: lot.doc.data().offer, driverId: driver});
-					//UNSUBSCRIBE - STOP LISTENING ON COMPONENT DID UNMOUNT
-				}
-			});
-		});
+		// await store.collection('lots').onSnapshot( allLots => { // This query needs to be majorly changed..
+		// 	allLots.docChanges().forEach(lot => {
+		// 		driver = lot.doc.data().driverId;
+		// 		//Not sure if needs another if statement but bid info should not changed unless its another bid
+		// 		if (lot.doc.data().passengerId === auth.currentUser.email && lot.doc.data().driverId !== null) {
+		// 			this.setState({showBid: true, offer: lot.doc.data().offer, driverId: driver});
+		// 			//UNSUBSCRIBE - STOP LISTENING ON COMPONENT DID UNMOUNT
+		// 		}
+		// 	});
+		// });
 
 		// This is the query that we want to make: It ensures that the button on MainScreen always displays properly
 		// Now it also makes sure that the receipt displays properly
 		let myPassengerLotHistory, mostRecentLotId;
         await store.collection("users").doc(auth.currentUser.email).get().then(user => {
 			this.setState({ currentLotId: user.data().currentLot.lotId })
             myPassengerLotHistory = user.data().myPassengerLotHistory
-        });
+		});
+
+		if (this.state.currentLotId) {
+			await store.collection("lots").doc(this.state.currentLotId).onSnapshot(lot => { // Where do we unsubscribe...?
```

Comment:
not sure how to unsub. i've tried it once on the other query but clearly it doesnt work since we still get a warning